src/ProjectOrigin.Registry.Server/Services/RabbitMqHttpClient.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using ProjectOrigin.Registry.Server.Interfaces;
+using ProjectOrigin.Registry.Server.Models;
+using ProjectOrigin.Registry.Server.Options;
+
+namespace ProjectOrigin.Registry.Server.Services;
+
+public sealed class RabbitMqHttpClient : IRabbitMqHttpClient, IDisposable
+{
+    private const string Protocol = "http";
+    private readonly HttpClient _httpClient;
```

Comment:
Would it make sense to use the HttpClientFactory here instead? 

Having a hard time figuring out if there is a captive dependency here

Reply:
Done :)

Reply:
Was actual thinking of https://learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-factory because it looks to me like this client is getting the same lifetime as the background service implementing the RabbitMqHttpClient and i believe this should be a short lived httpclient

https://blog.ploeh.dk/2014/06/02/captive-dependency/
https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines